### PR TITLE
Frontier

### PR DESCRIPTION
**Pull request by Nanway**
skkkkkrrrrrt skrrrrrrt
in this PR I

- Removed build/ devel files and added to gitignore so that we don't need to push as much code that we can just build in person anyway
- Added a gitignore
- Made turtlebot3 part of our repo instead of a submodule (as a submodule it's a nightmare cos we need to fork turtlebot3 and then start our own branch etc etc cos we need to modify a few things there for nav)
- Added a launch file for our roomba that will launch SLAM, move_base (navigation package) and explorer_lite, This will let the roomba start skrrt skrrting around the whole place

### CODE REVIEW

**Comment by Nanway** [comment]
@calvin1998  pls have a look before merging xoxo

File: src/m-explore/explore/include/explore/costmap_client.h [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  /**
+   * @brief Return a pointer to the "master" costmap which receives updates from
+   * all the layers.
+   *
+   * This pointer will stay the same for the lifetime of Costmap2DClient object.
+   */
+  const costmap_2d::Costmap2D* getCostmap() const
+  {
+    return &costmap_;
+  }
```
Comment (calvin1998) [inline src/m-explore/explore/include/explore/costmap_client.h:91]: double declaration

File: src/m-explore/explore/src/costmap_client.cpp [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+  std::string costmap_topic;
+  std::string footprint_topic;
+  std::string costmap_updates_topic;
+  param_nh.param("costmap_topic", costmap_topic, std::string("costmap"));
+  param_nh.param("costmap_updates_topic", costmap_updates_topic,
+                 std::string("costmap_updates"));
+  param_nh.param("robot_base_frame", robot_base_frame_,
+                 std::string("base_link"));
+  // transform tolerance is used for all tf transforms here
+  param_nh.param("transform_tolerance", transform_tolerance_, 0.3);
```
Comment (calvin1998) [inline src/m-explore/explore/src/costmap_client.cpp:64]: put topics in pubsubtopics.h with description of what they are

**Comment by calvin1998** [comment]
Which files do I look at @Nanway haha.

**Comment by calvin1998** [comment]
image

**Pull request merged**